addons/event_crm/models/event_registration.py

```diff
@@ -0,0 +1,333 @@
+            to_update.flush()  # compute notably partner-based fields if necessary
+            to_update.sudo()._update_leads(vals, lead_tracked_vals)
+
+        # handle triggers based on state
+        if not event_lead_rule_skip:
+            if vals.get('state') == 'open':
+                self.env['event.lead.rule'].search([('lead_creation_trigger', '=', 'confirm')]).sudo()._run_on_registrations(self)
+            elif vals.get('state') == 'done':
+                self.env['event.lead.rule'].search([('lead_creation_trigger', '=', 'done')]).sudo()._run_on_registrations(self)
+
+        return res
+
+    def _load_records_create(self, values):
+        """ In import mode: do not run rules those are intended to run when customers
+        buy tickets, not when bootstrapping a database. """
+        return super(EventRegistration, self.with_context(event_lead_rule_skip=True))._load_records_create(values)
+
+    def _load_records_write(self, values):
+        """ In import mode: do not run rules those are intended to run when customers
+        buy tickets, not when bootstrapping a database. """
+        return super(EventRegistration, self.with_context(event_lead_rule_skip=True))._load_records_write(values)
+
+    def _update_leads(self, new_vals, lead_tracked_vals):
+        """ Update leads linked to some registrations. Update is based depending
+        on updated fields, see ``_get_lead_contact_fields()`` and ``_get_lead_
+        description_fields()``. Main heuristic is
+
+          * check attendee-based leads, for each registration recompute contact
+            information if necessary (changing partner triggers the whole contact
+            computation); update description if necessary;
+          * check order-based leads, for each existing group-based lead, only
+            partner change triggers a contact and description update. We consider
+            that group-based rule works mainly with the main contact and less
+            with further details of registrations. Those can be found in stat
+            button if necessary.
+
+        :param new_vals: values given to write. Used to determine updated fields;
+        :param lead_tracked_vals: dict(registration_id, registration previous values)
+          based on new_vals;
+        """
+        for registration in self:
+            leads_attendee = registration.lead_ids.filtered(
+                lambda lead: lead.event_lead_rule_id.lead_creation_basis == 'attendee'
+            )
+            if not leads_attendee:
+                continue
+
+            old_vals = lead_tracked_vals[registration.id]
+            # if partner has been updated -> update registration contact information
+            # as they are computed (and therefore not given to write values)
+            if 'partner_id' in new_vals:
+                new_vals.update(**dict(
+                    (field, registration[field])
+                    for field in ('name', 'email', 'phone', 'mobile'))
```

Comment:
I know get_lead_contact_value also gives partner_id and partner_id is already set here, but maybe we could do `for field in contact_values`. To avoid duplicating same list of fields acoss the code